src/test/java/com/exasol/dbbuilder/dialects/exasol/ExasolDatabaseObjectCreationAndDeletionIT.java

```diff
@@ -298,4 +321,26 @@ void testInsertIntoTable() {
                     exception);
         }
     }
+
+    @Override
+    protected Matcher<DatabaseObject> existsInDatabase() {
+        return new ExistsInDatabaseMatcher();
+    }
+
+    private class ExistsInDatabaseMatcher extends AbstractDatabaseObjectCreationAndDeletionIT.ExistsInDatabaseMatcher {
```

Comment:
Could make this static class?

Reply:
I thought it's nicer that way since existsInDatabase() is just like a sentence. The performance loss is acceptable in test code... So what is the problem with a non-static class?

Reply:
It is one of the best practices. The main reasons:
> Without static member classes, each instance will have extra reference to its enclosing instance. This reference costs time and space, and result enclosing instance to be retained when it is eligible for garbage collection.
You can also read Item 22 in Effective Java.

Yes, in this test code maybe it is okay, though.

I think you can keep the method `existsInDatabase()` as is, and inside provide parameter to the new instance of the inner class.